File: backend/internal/models/report.go
package models

import "time"

type ReportSummary struct {
	TotalDevices   int64          `json:"total_devices"`
	OnlineDevices  int64          `json:"online_devices"`
	OfflineDevices int64          `json:"offline_devices"`
	LastOnline     *LastOnline    `json:"last_online,omitempty"`
	LastOffline    *LastOffline   `json:"last_offline,omitempty"`
}

type LastOnline struct {
	DeviceID     string    `json:"device_id"`
	DeviceName   string    `json:"device_name"`
	LastOnlineAt time.Time `json:"last_online_at"`
}

type LastOffline struct {
	DeviceID      string    `json:"device_id"`
	DeviceName    string    `json:"device_name"`
	LastOfflineAt time.Time `json:"last_offline_at"`
}

File: frontend/src/types/report.ts
export interface LastOnline {
    device_id: string;
    device_name: string;
    last_online_at: string;
}

export interface LastOffline {
    device_id: string;
    device_name: string;
    last_offline_at: string;
}

export interface ReportSummary {
    total_devices: number;
    online_devices: number;
    offline_devices: number;
    last_online?: LastOnline | null;
    last_offline?: LastOffline | null;
}